tests/test_examples.py

```diff
@@ -1186,6 +1191,333 @@ def test_example_get_current(self):
+    B_from_surf = surface_current_field2.compute_magnetic_field(
+        coords,
+        source_grid=LinearGrid(M=60, N=60, NFP=surface_current_field.NFP),
+        basis="rpz",
+    )
+    B_from_orig_surf = surface_current_field.compute_magnetic_field(
+        coords,
+        source_grid=LinearGrid(M=60, N=60, NFP=surface_current_field.NFP),
+        basis="rpz",
+    )
+    np.testing.assert_allclose(B, B_from_surf, atol=6e-3, rtol=1e-4)
+    np.testing.assert_allclose(B_from_orig_surf, B_from_surf, atol=5e-3, rtol=1e-4)
+
+    # test finding coils
+
+    numCoils = 15
+    with pytest.warns(UserWarning, match="intersecting"):
+        coilset2 = surface_current_field2.to_CoilSet(
+            num_coils=numCoils,
+        )
+    B_from_coils = coilset2.compute_magnetic_field(coords, basis="rpz")
+    np.testing.assert_allclose(B, B_from_coils, atol=7e-3, rtol=1e-3)
+
+
+@pytest.mark.regression
+@pytest.mark.solve
+@pytest.mark.slow
+def test_regcoil_helical_coils_check_B_pos_helicity(
+    regcoil_helical_coils_pos_helicity,
+):
+    """Test elliptical eq and winding surf helical coil regcoil solution."""
+    # with positive current helicity
+    (
+        initial_surface_current_field,
+        chi_B,
+        eq,
+    ) = regcoil_helical_coils_pos_helicity
+
+    surface_current_field = initial_surface_current_field.copy()
+
+    assert np.all(chi_B < 1e-5)
+    coords = eq.compute(["R", "phi", "Z", "B"])
+    B = coords["B"]
+    coords = np.vstack([coords["R"], coords["phi"], coords["Z"]]).T
+    B_from_surf = surface_current_field.compute_magnetic_field(
+        coords,
+        source_grid=LinearGrid(M=200, N=200, NFP=surface_current_field.NFP),
+        basis="rpz",
+    )
+    np.testing.assert_allclose(B, B_from_surf, atol=3e-3)
+
+
+@pytest.mark.regression
+@pytest.mark.solve
+@pytest.mark.slow
+def test_regcoil_helical_coils_check_coils_pos_helicity(
+    regcoil_helical_coils_pos_helicity,
+):
+    """Test elliptical eq and winding surf helical coil regcoil solution."""
+    # with positive current helicity
+    (
+        initial_surface_current_field,
+        _,
+        eq,
+    ) = regcoil_helical_coils_pos_helicity
+    surface_current_field = initial_surface_current_field.copy()
+
+    # test finding coils
+
+    numCoils = 15
+    with pytest.warns(UserWarning, match="intersecting"):
+        # the default grid for checking coil coil intersection
+        # is not good for long coils like these helical coils
+        coilset2 = surface_current_field.to_CoilSet(
+            num_coils=numCoils,
+            step=6,
+        )
+    assert not coilset2.is_self_intersecting(grid=LinearGrid(N=500))
+    coords = eq.compute(["R", "phi", "Z", "B"])
+    B = coords["B"]
+    coords = np.vstack([coords["R"], coords["phi"], coords["Z"]]).T
+    B_from_coils = coilset2.compute_magnetic_field(coords, basis="rpz")
+    np.testing.assert_allclose(B, B_from_coils, rtol=1e-3, atol=8e-3)
```

Comment:
Is atol large because of fewer numCoils?  Will it go down if I increase numCoils?

Reply:
Partly yes, there will be some floor though where the discretization error from making the current uniform along the coils is the dominant factor, then increasing num coils wont decrease the error anymore